### PR TITLE
organize shared production and dev props

### PR DESCRIPTION
**Pull request by ivansenic**
**What this PR does**:
Tries to organize the shared production/dev properties in one place (common project).

### CODE REVIEW

File: apis/sgv2-quarkus-common/src/main/resources/application.yaml [hunk 1]
```diff
@@ -152,3 +152,15 @@ quarkus:
               datastore:
                 configuration:
                   DataStorePropertiesConfiguration/fetchSupportedFeatures/Retry/maxRetries: 0
+
+# shared production properties
+"%prod":
```
Comment (ivansenic) [inline apis/sgv2-quarkus-common/src/main/resources/application.yaml:157]: I am actually not sure if I like the mixup of the production and development properties in one file.. I am used to have

* `application-dev.yaml`
* `application-prod.yaml` 

which clearly separates those.. The ``%dev` and `%prod` notation from quarkus is mixing everything in one file.. Should we change?

File: apis/sgv2-quarkus-common/src/main/resources/application.yaml [hunk 2]
```diff
@@ -152,3 +152,15 @@ quarkus:
...
+
+  quarkus:
+
+    # OpenTelemetry configuration
+    opentelemetry:
```
Comment (ivansenic) [inline apis/sgv2-quarkus-common/src/main/resources/application.yaml:162]: agreed to disable open telemetry by default in production @kathirsvn

File: apis/sgv2-quarkus-common/src/main/resources/application.yaml (hunk 1 above)
Comment (tatu-at-datastax) [reply apis/sgv2-quarkus-common/src/main/resources/application.yaml:157]: I agree that it gets bit messy. Assuming `application.yaml` has shared defaults, and overrides would be in `application-prod.yaml` (or -dev or -test, depending), that seems a bit better to me.
But I can live with either approach.

File: apis/sgv2-quarkus-common/src/main/resources/application.yaml (hunk 2 above)
Comment (tatu-at-datastax) [reply apis/sgv2-quarkus-common/src/main/resources/application.yaml:162]: How is it disabled here? Isn't this just setting sampler, but leaving it enabled (same as before change)?

Comment (ivansenic) [reply apis/sgv2-quarkus-common/src/main/resources/application.yaml:162]: It's not, we want to do it by default.. For now in prod it's disabled..

Comment (ivansenic) [reply apis/sgv2-quarkus-common/src/main/resources/application.yaml]: Actually, I can configure this without a `prod` profile at all..

Comment (tatu-at-datastax) [reply apis/sgv2-quarkus-common/src/main/resources/application.yaml]: Ah. I like this, cool.

**Review by tatu-at-datastax — APPROVED** [review]

**Pull request merged**